### PR TITLE
feat: Add kustomize tool

### PR DESCRIPTION
**Pull request by nikolaik**
This adds the [kustomize](https://github.com/kubernetes-sigs/kustomize) tool. This is needed to run `helmfile deps` in case a [helmfile uses kustomizations](https://github.com/roboll/helmfile/blob/master/docs/advanced-features.md#adhoc-kustomization-of-helm-charts).

I plan to open a PR to renovate to detect kustomization usage in a helmfile and install kustomize conditionally.

### CODE REVIEW

File: test/Dockerfile.jammy [hunk 1]
```diff
@@ -42,6 +42,9 @@ RUN install-tool flux v0.40.2
 # renovate: datasource=docker versioning=docker
 RUN install-tool golang 1.20.1
 
+# renovate: datasource=github-releases packageName=kubernetes-sigs/kustomize
+RUN install-tool kustomize v5.0.0
```
Comment (viceice) [inline test/Dockerfile.jammy:46]:
```suggestion
RUN install-tool kustomize 5.0.0
```
you already remove the v while extract version

File: test/latest/Dockerfile [hunk 2]
```diff
@@ -213,26 +213,37 @@ ARG VENDIR_VERSION=0.32.2
...
 ARG HELMFILE_VERSION=0.150.0
 
+# renovate: datasource=github-releases packageName=kubernetes-sigs/kustomize
+ARG KUSTOMIZE_VERSION=5.0.0
+
 RUN install-tool vendir "v${VENDIR_VERSION}"
 
 RUN install-tool helmfile "v${HELMFILE_VERSION}"
 
+RUN install-tool kustomize "v${KUSTOMIZE_VERSION}"
```
Comment (viceice) [inline test/latest/Dockerfile:223]:
```suggestion
RUN install-tool kustomize ${KUSTOMIZE_VERSION}
```

File: test/bash/tools/kustomize.bats [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  assert_output --partial "${TOOL_VERSION}"
+}
+
+@test "kustomize: link_tool" {
+  local TOOL_NAME=kustomize
+  local TOOL_VERSION
+  local bin_path=$(get_bin_path)
+
+  # renovate: datasource=github-releases packageName=kubernetes-sigs/kustomize
+  TOOL_VERSION=v5.0.0
```
Comment (viceice) [inline test/bash/tools/kustomize.bats:113]:
```suggestion
  TOOL_VERSION=5.0.0
```

File: renovate.json [hunk 4]
```diff
@@ -102,6 +103,11 @@
       "extends": ["schedule:weekly"],
       "automerge": true,
       "separateMinorPatch": false
+    },
+    {
+      "descriptions": "Handle tag prefix in kustomize monorepo",
+      "matchPackageNames": ["kubernetes-sigs/kustomize"],
```
Comment (viceice) [inline renovate.json:109]: please open a PR to the `.github` repo to add  this rule to the preset there.

Comment (nikolaik) [reply renovate.json:109]: Sure! https://github.com/containerbase/.github/pull/9

Should I remove it here then?

Comment (viceice) [reply renovate.json:109]: yes, remove it here

Comment (nikolaik) [reply renovate.json]: done 😊

File: src/usr/local/buildpack/tools/v2/kustomize.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+function install_tool () {
+  local versioned_tool_path
+  local file
+  local arch=linux_amd64
```
Comment (viceice) [inline src/usr/local/buildpack/tools/v2/kustomize.sh:6]: please prepare for arm64, checkout the open pr

File: test/bash/tools/kustomize.bats [hunk 6]
```diff
@@ -0,0 +1,155 @@
+setup_file () {
```
Comment (viceice) [inline test/bash/tools/kustomize.bats:1]: @Chumper can we somehow simplify the setup? looks like we've a lot duplicate stuff in that functions.

File: src/usr/local/buildpack/tools/v2/kustomize.sh (hunk 5 above)
Comment (nikolaik) [reply src/usr/local/buildpack/tools/v2/kustomize.sh:6]: nice, fixed!

File: test/bash/tools/kustomize.bats (hunk 6 above)
Comment (viceice) [reply test/bash/tools/kustomize.bats:1]: @nikolaik please remove the bats tests

File: src/usr/local/buildpack/tools/v2/kustomize.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  if [[ "$(uname -p)" = "aarch64" ]]; then
+    arch=linux_arm64
+  fi
+
+  local kustomize_file="${TOOL_NAME}_v${TOOL_VERSION}_${arch}.tar.gz"
```
Comment (viceice) [inline src/usr/local/buildpack/tools/v2/kustomize.sh:12]: i think you need to split declaration and assignment, otherwise shellcheck checks will fail

Comment (nikolaik) [reply src/usr/local/buildpack/tools/v2/kustomize.sh:12]: Hmm, I see multiple other usages like that, f.ex https://github.com/containerbase/base/blob/fc6b9dbdf86ceb8e05f16ad5100940ef9319a31c/src/usr/local/buildpack/tools/v2/maven.sh#L15

I can split them if needed ofc

**Comment by viceice** [comment]
please use the re-request review GitHub button when you're done for next round

**Comment by nikolaik** [comment]
> please use the re-request review GitHub button when you're done for next round

sorry, will do that in the future. Ready for re-review!

**Comment by viceice** [comment]
> > please use the re-request review GitHub button when you're done for next round
> 
> sorry, will do that in the future. Ready for re-review!

no worries, but without this i don't know when PR is ready for review again. alternative is to move the PR to draft and make it ready again so I get notifications.

**Review by viceice — APPROVED** [review]

**Pull request merged**